src/libratbag.h

```diff
@@ -1300,6 +1322,139 @@ enum ratbag_button_action_special
 ratbag_button_get_special(struct ratbag_button *button);
 
 /**
+ * @ingroup led
+ *
+ * LED mods.
+ */
+enum ratbag_led_mode {
+	RATBAG_LED_OFF = 0,
+	RATBAG_LED_ON = 1,
+	RATBAG_LED_CYCLE = 03,
+	RATBAG_LED_BREATH = 10
+};
+/**
+ * @ingroup led
+ *
+ * LED types, got only two types: Logo and Side.
+ */
+enum ratbag_led_type {
+	RATBAG_LED_TYPE_UNKNOWN = -1,
+	RATBAG_LED_TYPE_LOGO = 0,
+	RATBAG_LED_TYPE_SIDE
+};
+
+/**
+ * @ingroup led
+ *
+ * this function returns the mode for ratbag_led.
+ *
+ * @return The LED mod
+ *
+ * @see ratbag_led_set_mode
+ */
+enum ratbag_led_mode
+ratbag_led_get_mode(struct ratbag_led *led);
+/**
+ * @ingroup led
+ *
+ * this function returns the led color.
+ *
+ * @return The LED color
+ *
+ * @see ratbag_led_set_color
+ */
+struct ratbag_color
+ratbag_led_get_color(struct ratbag_led *led);
+/**
+ * @ingroup led
+ *
+ * this function returns the LED effect rate.
+ *
+ * @return The LED rate
+ *
+ * @see ratbag_led_set_rate
+ */
+int ratbag_led_get_rate(struct ratbag_led *led);
+/**
+ * @ingroup led
+ *
+ * this function returns the LED effect rate.
+ *
+ * @return The LED brightness
+ *
+ * @see ratbag_led_set_bright
+ */
+int ratbag_led_get_bright(struct ratbag_led *led);
+/**
+ * @ingroup led
+ *
+ * this function returns the LED object
+ *
+ * @return The LED rate
+ *
+ * @see ratbag_led_set_rate
+ */
+struct ratbag_led*
+ratbag_profile_get_led(struct ratbag_profile *profile,
+                       enum ratbag_led_type type);
+
+/**
+ * @ingroup led
+ *
+ * this function sets the LED mode.
+ *
+ * @param led A previously initialized ratbag LED
+ * @param mode LED mode.
+ * @return 0 on success or an error code otherwise.
+ *
+ * @see ratbag_led_get_mode
+ */
+enum ratbag_error_code
+ratbag_led_set_mode(struct ratbag_led *led, enum ratbag_led_mode mode);
+
+/**
+ * @ingroup led
+ *
+ * this function sets the LED color.
+ *
+ * @param led A previously initialized ratbag LED
+ * @param color LED color in hex (ffffff).
+ * @return 0 on success or an error code otherwise.
+ *
+ * @see ratbag_led_get_color
+ */
+enum ratbag_error_code
+ratbag_led_set_color(struct ratbag_led *led, struct ratbag_color color);
+
+/**
+ * @ingroup led
+ *
+ * this function sets the LED rate.
+ *
+ * @param led A previously initialized ratbag LED
+ * @param rate effect rate
+ * @return 0 on success or an error code otherwise.
+ *
+ * @see ratbag_led_get_rate
+ */
+enum ratbag_error_code
+ratbag_led_set_rate(struct ratbag_led *led, unsigned int rate);
+
+/**
+ * @ingroup led
+ *
+ * this function sets the LED bright.
```

Comment:
s/bright/brightness/ everywhere please. But one comment on the API documentation: it should be precise and describe what happens. Just saying "sets the LED brightness" isn't enough, there should be comments about allowed ranges, what is expected to happen on the device, etc. Please address this for all the new functions.